### PR TITLE
Added clustering github workflow

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by peusebiu** [comment]
need to change localstack with minio, as localstack seems to be very slow.

File: test/cluster/haproxy.cfg [hunk 1]
```diff
@@ -0,0 +1,46 @@
+global
```
Comment (rchincha) [inline test/cluster/haproxy.cfg:1]: move this file into examples/cluster/haproxy.cfg instead

Comment (rchincha) [reply test/cluster/haproxy.cfg:1]: Maybe also test i/o with `zb` also.

Comment (rchincha) [reply test/cluster/haproxy.cfg:1]: Also, give a separate name for this clustering setup - "stateless zot with shared reliable storage". There could be others and then we have to scramble.

Comment (peusebiu) [reply test/cluster/haproxy.cfg]: fixed

File: .github/workflows/cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          AWS_ACCESS_KEY_ID: minioadmin
+          AWS_SECRET_ACCESS_KEY: minioadmin
+      
+      - name: Run benchmark
+        run: |
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio1.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio2.json &
+            ./bin/zot-linux-amd64 serve test/cluster/config-minio3.json &
+            sleep 10
+            make -o run-minimal BENCH_OUTPUT=ci-cd run-bench
```
Comment (rchincha) [inline .github/workflows/cluster.yaml:111]: Why add a new Makefile target, just build bench and run it directly here.

**Comment by rchincha** [comment]
Also,

```
Complete requests:    0
Failed requests:      100
```

File: .github/workflows/cluster.yaml (hunk 2 above)
Comment (peusebiu) [reply .github/workflows/cluster.yaml:111]: I don't want to run minimal zot, but s3.

**Comment by rchincha** [comment]
@peusebiu, disable logs ("output": "/dev/null") so that the output is clean and readable.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**